cognite/client/_api/simulators/models_revisions.py

```diff
@@ -281,3 +290,32 @@ def create(
             items=items,
             input_resource_cls=SimulatorModelRevisionWrite,
         )
+
+    def retrieve_data(self, model_revision_external_id: str) -> SimulatorModelRevisionDataList:
+        """`Filter simulator model revision data <https://api-docs.cognite.com/20230101-alpha/tag/Simulator-Models/operation/get_simulator_model_revision_data_by_id>`_
+
+        Retrieves a list of simulator model revisions data that match the given criteria.
+
+        Args:
+            model_revision_external_id (str): The external id of the simulator model revision to filter by.
+        Returns:
+            SimulatorModelRevisionDataList: List of simulator model revision data
+
+        Examples:
+            List simulator model revision data:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.models.revisions.retrieve_data(model_revision_external_id="model_revision_1")
+        """
+        model_revisions_data_filter = SimulatorModelRevisionsDataFilter(
+            model_revision_external_id=model_revision_external_id,
+        )
+        self._warning.warn()
+
+        response = self._post(
+            url_path=f"{self._RESOURCE_PATH}/data/list",
+            headers={"cdf-version": "alpha"},
+            json={"items": [model_revisions_data_filter.dump()]},
+        )
```

Comment:
```suggestion
        self._warning.warn()

        response = self._post(
            url_path=f"{self._RESOURCE_PATH}/data/list",
            headers={"cdf-version": "alpha"},
            json={"items": [{"modelRevisionExternalId": model_revision_external_id}]},
        )
```
this is the only place `SimulatorModelRevisionsDataFilter` is used. we don't need it